### PR TITLE
Import FUKA initial data

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

On CaltechHPC load the `fuka/2023-08` module (in `module use /central/groups/sxs/modules`), then configure the CMake build (or set `FUKA_ROOT` yourself).

To use this set `InitialData` in a GHMHD input file to something like this:

```yaml
InitialData:
  FukaInitialData:
    InfoFilename: /central/groups/sxs/emost/fuka/codes/PythonTools/Example_id/converged_BNS_TOTAL.togashi.30.6.0.0.2.8.q1.0.0.09.info
    ElectronFraction: 0.15
```

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by nilsvu** [comment]
@ermost want to give this a try? When loading some sample ID in the FUKA repo I'm getting an FPE in Kadath, but that might be because my domain wasn't set up correctly or something like that.

File: src/IO/External/InterpolateFromFuka.cpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  auto& shift = get<gr::Tags::Shift<DataVector, 3>>(result);
+  get<0>(shift) = to_datavector(fuka_data[1]);
+  get<1>(shift) = to_datavector(fuka_data[2]);
+  get<2>(shift) = to_datavector(fuka_data[3]);
+  auto& spatial_metric = get<gr::Tags::SpatialMetric<DataVector, 3>>(result);
+  get<0, 0>(spatial_metric) = to_datavector(fuka_data[4]);
+  get<0, 1>(spatial_metric) = to_datavector(fuka_data[5]);
+  get<0, 2>(spatial_metric) = to_datavector(fuka_data[6]);
+  get<1, 1>(spatial_metric) = to_datavector(fuka_data[7]);
+  get<0, 2>(spatial_metric) = to_datavector(fuka_data[8]);
```
Comment (ermost) [inline src/IO/External/InterpolateFromFuka.cpp]: should this be 1,2?

It seems like you are not setting gamma_yz?

File: src/IO/External/InterpolateFromFuka.cpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  get<2, 2>(spatial_metric) = to_datavector(fuka_data[9]);
+  auto& extrinsic_curvature =
+      get<gr::Tags::ExtrinsicCurvature<DataVector, 3>>(result);
+  get<0, 0>(extrinsic_curvature) = to_datavector(fuka_data[10]);
+  get<0, 1>(extrinsic_curvature) = to_datavector(fuka_data[11]);
+  get<0, 2>(extrinsic_curvature) = to_datavector(fuka_data[12]);
+  get<1, 1>(extrinsic_curvature) = to_datavector(fuka_data[13]);
+  get<0, 2>(extrinsic_curvature) = to_datavector(fuka_data[14]);
```
Comment (ermost) [inline src/IO/External/InterpolateFromFuka.cpp]: should this be 1,2?

It seems like you are not setting K_yz?

**Review by ermost — CHANGES_REQUESTED** [review]

**Comment by ermost** [comment]
Sure, happy to take a look.

File: src/IO/External/InterpolateFromFuka.cpp (hunk 1 above)
Comment (nilsvu) [reply src/IO/External/InterpolateFromFuka.cpp]: Thanks, fixed! The FPE happens in Kadath though, so I don't think this was it..

Also I'd like to add a test for this that is enabled only when FUKA is available. Maybe we could put some sample data into the FUKA install directory? Do you think some of the data in the FUKA repo would be appropriate, or should we generate new data?

**Comment by nilsvu** [comment]
This is ready for review now.

**Comment by nilsvu** [comment]
I successfully evolved some FUKA BNS ID with this. Here's the initial frame:

Bildschirmfoto 2023-09-27 um 22 48 42

**Comment by nilsvu** [comment]
Can't fix the remaining clang-tidy warnings I think.

File: cmake/FindFUKA.cmake [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Find the FUKA initial data code (https://bitbucket.org/fukaws/fuka/src/fuka/)
+#
+# Pass `FUKA_ROOT` to the CMake build configuration to set up the following
```
Comment (nilsdeppe) [inline cmake/FindFUKA.cmake:6]: maybe dox to our build system md file?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM, please squash

File: src/PointwiseFunctions/AnalyticData/GrMhd/FukaInitialData.hpp [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+class FukaInitialData : public evolution::initial_data::InitialData,
+                        public evolution::NumericInitialData,
+                        public AnalyticDataBase {
+ public:
+  struct InfoFilename {
+    using type = std::string;
+    static constexpr Options::String help = {
+        "Path to the '.info' file of data produced by FUKA."};
+  };
+
```
Comment (ermost) [inline src/PointwiseFunctions/AnalyticData/GrMhd/FukaInitialData.hpp:50]: Should we add a comment here or in the help string that the constant electron fraction option is most of the time inconsistent with the ID, and will need to be replaced later?

Comment (nilsdeppe) [reply src/PointwiseFunctions/AnalyticData/GrMhd/FukaInitialData.hpp:50]: Yes, let's do that! Does the ID write out consistent a electron fraction? If not, should that be added? We've talked about this for SpEC too, and that's the plan for spec

Comment (nilsvu) [reply src/PointwiseFunctions/AnalyticData/GrMhd/FukaInitialData.hpp:50]: Neither electron fraction nor temperature are written out by FUKA. I added a note to the dox for this class and opened https://github.com/sxs-collaboration/spectre/issues/5518.

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**